File: TownOfSalemCalculator/src/townofsalemcalculator/Conditions/Concrete/KillMessages/VigilanteKill.java
package townofsalemcalculator.Conditions.Concrete.KillMessages;

import java.util.Arrays;
import java.util.List;
import static townofsalemcalculator.Role.Vigilante;

/**
 * The Concrete Condition that a Vigilante has killed someone
 * @author Multifacio
 * @version 1.1
 * @since 2017-1-29
 */
public class VigilanteKill extends KillMessage {
    public VigilanteKill() {
        super(Vigilante);
    }
    
    @Override
    public String getCondition() {
        return "He was shot by a Vigilante.";
    }

    @Override
    public List<String> keyWords() {
        List<String> keyWords = Arrays.asList(new String[]{"He", "Died", "Killed", "Death", "Shot", "Shoot"});
        keyWords.addAll(Vigilante.getKeyWords());
        return keyWords;
    }

}

File: src/Conditions/Concrete/KillTypes/VigilanteKill.py
from src.Concepts.Role import Role
from src.Conditions.Concrete.ExistenceCondition import ExistenceCondition
from typing import FrozenSet

class VigilanteKill(ExistenceCondition):
    """ The Vigilante Kill condition is used if some player was killed by a Vigilante. """

    def __init__(self, amnesiacRemembered: FrozenSet[Role] = None):
        super().__init__(frozenset({Role.VIGILANTE}), amnesiacRemembered)